File: JavaScript/ChangeBit.js
let character = 'A';
let offset = 5;

let asciiValue = character.charCodeAt(0);
let binaryConversion = []

while(asciiValue>0) {
    let remainder = asciiValue%2;
    asciiValue = Math.trunc(asciiValue/2); 
    binaryConversion.push(remainder);
}

if(offset>binaryConversion.length) {
    let extraBit = offset-binaryConversion.length;
    for(let i=0; i<extraBit; i++) {
        binaryConversion.push(0);
    }
    binaryConversion.push(1); 
} else if(offset<binaryConversion.length) {
    if(binaryConversion[offset-1]==0)
        binaryConversion[offset-1]=1;
    else if(binaryConversion[offset-1]==1)
        binaryConversion[offset-1]=0;
}

let sum = 0;
for(let i=0; i<binaryConversion.length; i++) {
    sum += Math.pow(2,i)*binaryConversion[i];
}

//console.log(binaryConversion);
console.log(String.fromCharCode(sum));

File: ProgrammingBasics/ChangeBit.c
#include<stdio.h>
#include<Math.h>

int flipBit(int num, int offset)
{
	int remainder;
	int BinaryArray[100] = {0};
	int i=0;
	while(num>0)
	{
		remainder = num%2;
		num = num/2;
		BinaryArray[i++] = remainder;
	}
	
	if(BinaryArray[offset-1]==1)
		BinaryArray[offset-1]=0;
	else
		BinaryArray[offset-1]=1;
		
	int sum=0;
	for(i=0; i<(sizeof(BinaryArray)/sizeof(BinaryArray[0])); i++)
	{
		sum += ((int)pow(2,i))*BinaryArray[i];
	}
	return sum;
}

const char* testFunction(char character, int offset, int expectedResult) 
{
    if((char)flipBit(character,offset)==expectedResult)
        return "Pass";
    else   
        return "Not pass";
}

int main()
{
	/*char character;
	int num;
	printf("\nEnter Character: ");
	scanf("%c",&character);

	printf("Enter offset number: ");
	scanf("%d",&num);

	flipBit(character, num);*/

	printf("Test Cases:\n");
	printf("\nCharacter\tOffset\tExpected Result\tActual Result\tTest Case");
	printf("\nA\t\t5\t%c\t\tQ\t\t%s",flipBit('A',5),testFunction('A',5,'Q'));
	printf("\na\t\t3\t%c\t\te\t\t%s",flipBit('a',3),testFunction('a',3,'e'));
	printf("\nZ\t\t4\t%c\t\tR\t\t%s",flipBit('Z',4),testFunction('Z',4,'R'));
}
